### ISSUE TITLE
Error:  Resource temporarily unavailable

### ISSUE THREAD

**Issue by MrBLJ** [Jackett/Jackett#5075]
Hi all,

I'm currently facing an issue with Jackett. It's similar to issue #4947 and I've already searched for solutions by myslef. Unfortunately, so far, I'm stuck with this situation.

This is a tracker related issue. All queries/test I run on my trackers give me the followong result: Resource temporarily unavailable.
At first, I thought it was related to pi-hole since I started to use it recently, but after disabling (even uninstalling) it, the issue remains.

I'm using the docker image for my home server (mainly for radarr, sonarr & lidarr).

-------------------------------
If you are experiencing an issue with a tracker, then:
1. Use your browser to check you can access the site directly, and if a login is required, check you can login and that you do not have any outstanding account issues.
**Done, I can access the sites directly.**

2. If you haven't already, try upgrading to the latest version of Jackett, your issue may have already been resolved.
**I'm using 0.11.224.0. I think it's the latest available version of the docker image.**

3. If it is still not working for you, then a **full enhanced log must be included**. Instructions for obtaining logs are here: https://github.com/Jackett/Jackett#troubleshooting
`Jackett.Common.IndexerException: Exception (yts): Resource temporarily unavailable ---> System.Net.Http.HttpRequestException: Resource temporarily unavailable ---> System.Net.Sockets.SocketException: Resource temporarily unavailable
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Jackett.Common.Indexers.BaseWebIndexer.RequestStringWithCookiesAndRetry(String url, String cookieOverride, String referer, Dictionary`2 headers) in /home/appveyor/projects/jackett/src/Jackett.Common/Indexers/BaseIndexer.cs:line 465
   at Jackett.Common.Indexers.Yts.PerformQuery(TorznabQuery query, Int32 attempts) in /home/appveyor/projects/jackett/src/Jackett.Common/Indexers/yts.cs:line 115
   at Jackett.Common.Indexers.Yts.PerformQuery(TorznabQuery query) in /home/appveyor/projects/jackett/src/Jackett.Common/Indexers/yts.cs:line 71
   at Jackett.Common.Indexers.BaseIndexer.ResultsForQuery(TorznabQuery query) in /home/appveyor/projects/jackett/src/Jackett.Common/Indexers/BaseIndexer.cs:line 331
   --- End of inner exception stack trace ---
   at Jackett.Common.Indexers.BaseIndexer.ResultsForQuery(TorznabQuery query) in /home/appveyor/projects/jackett/src/Jackett.Common/Indexers/BaseIndexer.cs:line 351
   at Jackett.Common.Indexers.BaseWebIndexer.ResultsForQuery(TorznabQuery query) in /home/appveyor/projects/jackett/src/Jackett.Common/Indexers/BaseIndexer.cs:line 815
   at Jackett.Common.Services.IndexerManagerService.TestIndexer(String name) in /home/appveyor/projects/jackett/src/Jackett.Common/Services/IndexerManagerService.cs:line 222
   at Jackett.Server.Controllers.IndexerApiController.Test() in /home/appveyor/projects/jackett/src/Jackett.Server/Controllers/IndexerApiController.cs:line 128
   at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeNextActionFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
   at Microsoft.AspNetCore.Builder.RouterMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)
   at Jackett.Server.Middleware.CustomExceptionHandler.Invoke(HttpContext httpContext) in /home/appveyor/projects/jackett/src/Jackett.Server/Middleware/CustomExceptionHandler.cs:line 29`

**Comment by garfield69** [comment]
for YTS the jackett indexer does not scrape the HTML from the web site yts.am directly.
instead it uses the API provided by the site at https://yts.am/api/v2/list_movies.json

the temporary unavailable message is from this service, which was unavailable, presumably because of maintenance or other reason.
 
I just tried it now and the service is back up.

**Comment by MrBLJ** [comment]
Hi,

Thanks for your answer. I should have been more accurate in my description, but the error I mentionned appeéars for **all** indexers, not just YTS.

I tried to apply the fix you suggested, without any success. All indexers still give me _Error: Resource temporarily unavailable_.

May I suggest that the issue remains open for the time being ?

**Comment by garfield69** [comment]
@flightlevel @kaso17  any thoughts on what this might be about? 
rather strange error message to be getting across the board, but then again, its docker, so I guess we shouldn't be surprised.

**Comment by MrBLJ** [comment]
Updated to jackett v0.11.228.0. Issue remains, for _all_ indexers.
I can access all indexers websites directly, as before.

I'm looking into my pihole log, but I don't see any blocked queries or whatsoever. Note that the issue persists even with pihole turned off.

**Comment by flightlevel** [comment]
Try the latest Jackett release v0.11.230
If you are still getting this problem:
From a terminal run `docker exec -it jackett /bin/bash` replace `jackett` with whatever you called your Jackett container. You now have bash access to container
run this command for a few different urls, e.g google and a couple of trackers and post the output
`curl -vvv https://www.google.com/`

- What CPU and how much RAM?
- How many indexers do you have?
- Are you limiting the docker containers memory or cpu?
- Are you using the aggregate `/all` indexer?
- Are you using a proxy?
-If you restart the container and test just one indexer from the Jackett dashboard, does it work?

**Comment by MrBLJ** [comment]
Problem solved !

Since you took some time to help me, I'll answer your questions first:

- I'm using a NUC i5 with 4Gb of ram.
- I have 17 indexers.
- There's no limitation for CPU/memory (see below for remarks about that).
- I'm using the aggregate /all indexer.
- I'm using traefik as a reverse proxy.
- Using just one indexer did not work.

**Solution:**
I tried what you suggested and could not ping anything successfuly (failure in name resolution). Then I realized something must be wrong with my DNS config. [The fix can be found here](https://development.robinwinslow.uk/2016/06/23/fix-docker-networking-dns/). It worked flawlessly.

Sidenote: I noticed that  everything was running quite slowly for a few days. I had a look at what was using memory and behold, _jackett_ was using an unholy amount of ram (~60% of it). When I fixed the DNS related issue, the memory usage dropped significantely. Note sure this is of any use to you but in any case, I thought you'd might be interested.

I'll make sure to set memory usage limitations in the future.
